maestro-ios-driver/src/main/kotlin/util/LocalSimulatorUtils.kt

```diff
@@ -171,30 +177,112 @@ object LocalSimulatorUtils {
             .waitFor()
     }
 
+    private fun isAppRunning(deviceId: String, bundleId: String): Boolean {
+        val process = ProcessBuilder(
+            listOf(
+                "xcrun",
+                "simctl",
+                "spawn",
+                deviceId,
+                "launchctl",
+                "list",
+            )
+        ).start()
+
+        return String(process.inputStream.readBytes()).trimEnd().contains(bundleId)
+    }
+
+    private fun ensureStopped(deviceId: String, bundleId: String) {
+        MaestroTimer.withTimeout(10000) {
+            while (true) {
+                if (isAppRunning(deviceId, bundleId)) {
+                    Thread.sleep(1000)
+                } else {
+                    return@withTimeout
+                }
+            }
+        } ?: throw SimctlError("App $bundleId did not stop in time")
+    }
+
+    private fun ensureRunning(deviceId: String, bundleId: String) {
+        MaestroTimer.withTimeout(10000) {
+            while (true) {
+                if (isAppRunning(deviceId, bundleId)) {
+                    return@withTimeout
+                } else {
+                    Thread.sleep(1000)
+                }
+            }
+        } ?: throw SimctlError("App $bundleId did not start in time")
+    }
+
+    private fun copyDirectoryRecursively(source: Path, target: Path) {
+        Files.walk(source).forEach { path ->
+            val targetPath = target.resolve(source.relativize(path).toString())
+            if (Files.isDirectory(path)) {
+                Files.createDirectories(targetPath)
+            } else {
+                Files.copy(path, targetPath)
+            }
+        }
+    }
+
+    private fun deleteFolderRecursively(folder: File): Boolean {
+        if (folder.isDirectory) {
+            folder.listFiles()?.forEach { child ->
+                deleteFolderRecursively(child)
+            }
+        }
+        return folder.delete()
+    }
+
+    private fun reinstallApp(deviceId: String, bundleId: String) {
+        val pathToBinary = Path(getAppBinaryDirectory(deviceId, bundleId))
+
+        if (Files.isDirectory(pathToBinary)) {
+            val tmpDir = createTempDirectory()
+            val tmpBundlePath = tmpDir.resolve("$bundleId-${System.currentTimeMillis()}.app")
+
+            logger.info("Copying app binary from $pathToBinary to $tmpBundlePath")
+            Files.copy(pathToBinary, tmpBundlePath)
+            copyDirectoryRecursively(pathToBinary, tmpBundlePath)
+
+            logger.info("Reinstalling and launching $bundleId")
+            uninstall(deviceId, bundleId)
+            install(deviceId, tmpBundlePath)
+            launch(deviceId, bundleId, emptyList(), null)
+
+            ensureRunning(deviceId, bundleId)
```

Comment:
```suggestion
```
I tested this without launching the app locally, and it works as expected.

Reasoning:
clear state can be used either as a standalone command (`- clearState`) or as part of the `- launchApp` command. In the former case, we don't want to launch the app because it's not specified by the command. In the latter case, the app will be launched automatically after clearState, so there's no need to relaunch it.

WDYT?

Reply:
Good point, I missed that `- clearState` does not automatically launch the app again 👍